src/stim/simulators/frame_simulator.pybind.cc

```diff
@@ -820,6 +820,14 @@ void stim_pybind::pybind_frame_simulator_methods(
 
             size_t major = arr.shape(0);
             size_t minor = arr.shape(1);
+
+            if (minor != self.batch_size) {
+                throw std::invalid_argument(
+                "broadcast_pauli_errors can only accept mask that has minor shape equal to the batch_size");
+            }
+
+            self.ensure_safe_to_do_circuit_with_stats(CircuitStats{.num_qubits=(uint32_t)major});
```

Comment:
This cast isn't safe. What if the shape is larger than 2^32? Do a bounds check first. Gotta make sure we never segfault people's python instances in their notebooks; they hate that.